### PR TITLE
[Refactor](Metadata tvf) Reconstruct Metadata table-value function into a more general framework.

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -717,16 +717,16 @@ struct TInitExternalCtlMetaResult {
 
 enum TSchemaTableName{
```
Comment (morningman) [inline gensrc/thrift/FrontendService.thrift:718]:
```suggestion
enum TSchemaTableName {
```

File: be/src/vec/exec/scan/vmeta_scanner.cpp [hunk 2]
```diff
@@ -158,38 +153,48 @@ Status VMetaScanner::_fill_block_with_remote_data(const std::vector<MutableColum
...
+    VLOG_CRITICAL << "VMetaScanner::_fetch_iceberg_metadata";
+    if (!meta_scan_range.__isset.iceberg_params) {
+        return Status::InternalError("Can not find TIcebergMetadataParams from meta_scan_range.");
+    }
+    // create request
     TFetchSchemaTableDataRequest request;
     request.cluster_name = "";
     request.__isset.cluster_name = true;
-    request.schema_table_name = TSchemaTableName::ICEBERG_TABLE_META;
+    request.schema_table_name = TSchemaTableName::METADATA_TABLE;
```
Comment (morningman) [inline be/src/vec/exec/scan/vmeta_scanner.cpp:177]: use `request.set_xxx()` methods

File: be/src/vec/exec/scan/vmeta_scanner.cpp [hunk 3]
```diff
@@ -158,38 +153,48 @@ Status VMetaScanner::_fill_block_with_remote_data(const std::vector<MutableColum
...
 
     TNetworkAddress master_addr = ExecEnv::GetInstance()->master_info()->network_address;
     TFetchSchemaTableDataResult result;
-
     RETURN_IF_ERROR(ThriftRpcHelper::rpc<FrontendServiceClient>(
             master_addr.hostname, master_addr.port,
             [&request, &result](FrontendServiceConnection& client) {
                 client->fetchSchemaTableData(result, request);
             },
             config::txn_commit_rpc_timeout_ms));
```
Comment (morningman) [inline be/src/vec/exec/scan/vmeta_scanner.cpp:197]: It is strange to use `txn_commit_rpc_timeout_ms` as the timeout of meta scanner

File: be/src/vec/exec/scan/vmeta_scanner.cpp [hunk 4]
```diff
@@ -158,38 +153,48 @@ Status VMetaScanner::_fill_block_with_remote_data(const std::vector<MutableColum
...
+    case TMetadataType::ICEBERG:
+        return _fetch_iceberg_metadata(meta_scan_range);
+    default:
+        _meta_eos = true;
+        break;
+    }
+    return Status::OK();
+}
+
+Status VMetaScanner::_fetch_iceberg_metadata(const TMetaScanRange& meta_scan_range) {
```
Comment (morningman) [inline be/src/vec/exec/scan/vmeta_scanner.cpp:168]: the RPC logic is identical for all kinds of metadata fetch request.
I think it should be extracted to a method.
And for different request type, use different method named as `buildXXXRequest()`

Comment (BePPPower) [reply be/src/vec/exec/scan/vmeta_scanner.cpp:168]: done

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by BePPPower** [comment]
run p0

**Comment by BePPPower** [comment]
run p0

**Comment by BePPPower** [comment]
run p0

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**